### PR TITLE
Spark: Fail on recursive cycle in view

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckViews.scala [hunk 1]
```diff
@@ -63,4 +72,29 @@ object CheckViews extends (LogicalPlan => Unit) {
...
   }
+
+  private def checkCyclicViewReference(
+    viewIdent: Seq[String],
+    plan: LogicalPlan): Unit = {
+    plan match {
+      case sub@SubqueryAlias(_, Project(_, _)) =>
+        val ident: Seq[String] = sub.identifier.qualifier :+ sub.identifier.name
+        if (ident == viewIdent) {
+          throw new AnalysisException(String.format("Recursive cycle in view detected: %s", viewIdent.asIdentifier))
```
[the remark this answers is not in the document]
Comment (amogh-jahagirdar) [reply spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckViews.scala:83]: Do you mean it includes the whole cycle path? If so I think that would be useful for someone trying to debug why this error is happening. Basically as we recurse on the cycle check, we can keep track of the path of identifiers seen by appending the identifier that's being visited currently.

[the remark this answers is not in the document]
Comment (nastra) [reply spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckViews.scala]: @amogh-jahagirdar I've added the cycle to the error msg in the last commit. Can you take another look please?

**Comment by nastra** [comment]
CI failure is known and related to https://github.com/apache/iceberg/issues/10038

**Review by amogh-jahagirdar — APPROVED** [review]
Thanks for fixing this, this looks good to me!

**Pull request merged**